client/src/pages/treedata/CarbonCalculator.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable max-len */
+/* eslint-disable react/jsx-one-expression-per-line */
+import React from 'react';
+import DataTable from './DataTable';
+import { format } from 'date-fns';
+
+const carbonCalculatorExplainer = (carbonDetails) => ({
+  weightTotalGreen: `1.2 * weightaboveGround = 1.2 * ${carbonDetails.weightaboveGround} = ${carbonDetails.weightTotalGreen} lbs`,
+  weightDry: `0.725 * weightTotalGreen = 0.725 * ${carbonDetails.weightTotalGreen} = ${carbonDetails.weightDry} lbs`,
+  weightCarbon: ` 0.5 * weightDry = 0.5 * ${carbonDetails.weightDry} = ${carbonDetails.weightCarbon} lbs`,
+  weightCarbonDioxide: `3.67 * weightCarbon  = 3.67 * ${carbonDetails.weightCarbon} = ${carbonDetails.weightCarbonDioxide} lbs CO2 sequestered in ${carbonDetails.age} years`,
+  weightCarbonDioxidePerYear: `weightCarbonDioxide / age  = ${carbonDetails.weightCarbonDioxide} / ${carbonDetails.age} = ${carbonDetails.weightCarbonDioxidePerYear} lbs CO2 sequestered per years`,
+  weightCarbonDioxideInKg: `${carbonDetails.convertWeightCarbonDioxideToKg} kg`,
+});
+
+const carbonCalculatorMath = (age, dbh, height, planted, decimalYear) => ({
+  planted: format(new Date(planted), 'yyyy-MM-dd'),
+  decimalYear,
+  DBH: dbh,
+  height,
+  weightaboveGround: 0.25 * dbh * height,
+  weightTotalGreen: 1.2 * 0.25 * dbh * height,
+  weightDry: 0.725 * 1.2 * 0.25 * dbh * height,
+  weightCarbon: 0.5 * 0.725 * 1.2 * 0.25 * dbh * height,
+  weightCarbonDioxideInLbs: 3.67 * 0.5 * 0.725 * 1.2 * 0.25 * dbh * height,
+  weightCarbonDioxideInKg: 0.45359237 * 3.67 * 0.5 * 0.725 * 1.2 * 0.25 * dbh * height,
+  weightCarbonDioxidePerYearInLbs: (3.67 * 0.5 * 0.725 * 1.2 * 0.25 * dbh * height) / age,
+  weightCarbonDioxidePerYearInKg: (0.45359237 * 3.67 * 0.5 * 0.725 * 1.2 * 0.25 * dbh * height) / age,
```

Comment:
Maybe define each of these as constants to use in the next calculation, so the same terms aren't repeated over and over, e.g., `weightCarbonDioxideInKg: 0.45359237 * weightCarbonDioxideInLbs`.

Reply:
I'm not sure what's clearest for the tree people.  At first glance, I assumed these were independent calculations, and then I saw they were steps in one over all calculation.  If this does need to be tweaked, it would be easier to change in one place than making sure each 1.2 was changed to 1.15, or whatever.